### ISSUE TITLE
Console.CursorSize method does not work in terminal but works in conhost.

### ISSUE THREAD

**Issue by stevewpalmer** [microsoft/terminal#16027]
### Windows Terminal version

1.17.11461.0

### Windows build number

10.0.22621.2283

### Steps to reproduce

1. Compile the sample code at https://learn.microsoft.com/en-us/dotnet/api/system.console.cursorsize?view=net-7.0.

2. Run under Terminal. Note that the cursor size remains what was set in the profile.

3. Run under conhost.exe (the old cmd.exe). Note that the cursor size changes in response to changes to the CursorSize property.

### Expected Behavior

I expected the cursor size to change in response to changes to the property as the documentation for CursorSize says nothing about the functionality being deprecated or impaired under Terminal.

### Actual Behavior

The cursor size setting in the terminal profile remains constant and overrides any change to the CursorSize property.

**Comment by lhecker** [comment]
I believe this issue is a duplicate of #7382.
tl;dr: We need to define a custom escape sequence first that communicates from ConPTY (the console API -> VT translator) to Windows Terminal that the cursor size has changed.
